stable_baselines3/common/buffers.py

```diff
@@ -310,24 +310,27 @@ def compute_returns_and_advantage(self, last_value: th.Tensor, dones: np.ndarray
         where R is the discounted reward with value bootstrap,
         set ``gae_lambda=1.0`` during initialization.
 
-        :param last_value:
+        :param last_values:
         :param dones:
 
         """
         # convert to numpy
-        last_value = last_value.clone().cpu().numpy().flatten()
+        last_values = last_values.clone().cpu().numpy().flatten()
 
         last_gae_lam = 0
+        # Last timestep: use the last predicted value for bootstrap
+        # step == self.buffer_size - 1
+        next_non_terminal = 1.0 - dones
+        next_values = last_values
         for step in reversed(range(self.buffer_size)):
```

Comment:
I think the "optimization" sought here is to move the first iteration completely outside the loop (i.e. do delta, last_gae_lam and self.advantages steps). This would remove if clause completely and potentially simplify code, but also seem messy with copy-pasted parts, so up to you if you want to include.

Reply:
I will probably revert back then

Reply:
I don't know, what do you think? (I don't like duplicating code and as you mentioned I don't think we will get any speedup, I did it mostly for clarity)

Reply:
Fair enough, I was on the edge which would be the way go (clarity/cleanity over last bit of performance).